### PR TITLE
feat: validator volume increase test

### LINKED ISSUE #120: Tests for validator
### Based on #99 
It is required that we can statically test the codebase of the validator

- Define validator scope
- Tests for utils
- Tests for Rewards calculations
- Tests for Weights calculations
- Tests for miners ( UIDs ) selection

### PR DESCRIPTION
**Pull request by hide-on-bush-x**
**Description**

#### Add Integration Tests for Validator Server
This PR introduces a suite of integration tests for the Validator server to ensure its endpoints are functioning correctly. The tests are implemented using the `unittest` framework and cover the following scenarios:

1. **Server Health Check**:
   - **Test Method**: `test_server_running`
   - **Description**: Verifies that the server is running and accessible via the `/healthcheck` endpoint.
   - **Expected Outcome**: The server should return a status code of 200.

2. **Fetch Recent Tweets**:
   - **Test Method**: `get_recent_tweets`
   - **Description**: Checks if the server can fetch recent tweets from the `/data/twitter/tweets/recent` endpoint.
   - **Expected Outcome**: The server should return a status code of 200 and a non-empty list of tweets.

3. **Fetch Volumes**:
   - **Test Method**: `get_volumes`
   - **Description**: Ensures the server can retrieve volume data from the `/volumes` endpoint.
   - **Expected Outcome**: The server should return a status code of 200 and a non-empty list of volumes.

4. **Volumes Change After Fetching Recent Tweets**:
   - **Test Method**: `test_volumes_change_after_getting_recent_tweets`
   - **Description**: Validates that the volume data changes after fetching recent tweets, provided there are tweets available.
   - **Expected Outcome**: The `tempo` field in the latest volume should remain the same, while the `miners` field should change after fetching recent tweets.

#### Additional Notes:
- The tests include assertions to ensure the server responses are as expected.
- The `time.sleep(15)` is used to introduce a delay before re-checking the volumes to allow for any asynchronous updates.
- The test methods `get_recent_tweets` and `get_volumes` are designed to be reusable for other test cases.

#### How to Run:
- Ensure the Validator server is running locally on `http://localhost:8000`.
- Execute the tests using the command:
  ```bash
  python -m unittest tests/validator.py
  ```

This PR aims to enhance the reliability of the Validator server by ensuring its critical endpoints are tested and verified.

This PR fixes #120 

**Notes for Reviewers**
Testing twitter flow behaivour, checking volume fluctuations

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: tests/validator.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import requests
+
+
+class ValidatorServerTestCase(unittest.TestCase):
+    """
+    This class contains integration tests for the Validator server.
+    """
+
+    def setUp(self):
+        self.base_url = "http://localhost:8000"
```
Comment (mudler) [inline tests/validator.py:11]: mmh maybe better get this from an environment variable and default to localhost:8000 - just thinking that might come more handy when trying to run tests over Github actions

File: tests/validator.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            if len(tweets) > 0:
+                success = True
+            else:
+                success = False
+        else:
+            self.assertTrue(False, f"Response status code: {response.status_code}")
+
+        self.assertTrue(success, "No tweets found in fetching recent tweets")
+
+        self.assertEqual(
```
Comment (mudler) [inline tests/validator.py:43]: should we also maybe assert here that we get some data as well from the JSON response? e.g. at least 1 tweet should be returned

File: tests/validator.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        initial_latest_tempo = initial_volumes[-1]["tempo"]
+        initial_miners = initial_volumes[-1]["miners"]
+        print("Initial volumes:", initial_volumes)
+
+        # Fetch recent tweets
+        tweets = self.get_recent_tweets()
+        tweets_length = len(tweets)
+        print("Tweets length:", tweets_length)
+
+        # Wait for 2 seconds before checking volumes again
```
Comment (mudler) [inline tests/validator.py:96]:
```suggestion
        # Wait for few seconds before checking volumes again
```

File: tests/validator.py (hunk 2 above)
Comment (hide-on-bush-x) [reply tests/validator.py:43]: Thats happening!
On line 34 `response.json()` is the list of tweets, if that list's length is greater than 0 `success = True`

then if `success == False` => `No tweets found in fetching recent tweets`

Comment (mudler) [reply tests/validator.py:44]: ouch sorry, my bad!

**Pull request merged**